File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-fx-5-4/UpperBound.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class UpperBound : ISerialized
    {
        public UpperBound(XmlNode xmlNode)
        {
            XmlNode maximumInclusiveNode = xmlNode.SelectSingleNode("maximumInclusive");
            
            if (maximumInclusiveNode != null)
            {
                if (maximumInclusiveNode.Attributes["href"] != null || maximumInclusiveNode.Attributes["id"] != null) 
                {
                    if (maximumInclusiveNode.Attributes["id"] != null) 
                    {
                        maximumInclusiveIDRef_ = maximumInclusiveNode.Attributes["id"].Value;
                        PositiveDecimal ob = new PositiveDecimal(maximumInclusiveNode);
                        IDManager.SetID(maximumInclusiveIDRef_, ob);
                    }
                    else if (maximumInclusiveNode.Attributes["href"] != null)
                    {
                        maximumInclusiveIDRef_ = maximumInclusiveNode.Attributes["href"].Value;
                    }
                    else
                    {
                        maximumInclusive_ = new PositiveDecimal(maximumInclusiveNode);
                    }
                }
                else
                {
                    maximumInclusive_ = new PositiveDecimal(maximumInclusiveNode);
                }
            }
            
        
            XmlNode maximumExclusiveNode = xmlNode.SelectSingleNode("maximumExclusive");
            
            if (maximumExclusiveNode != null)
            {
                if (maximumExclusiveNode.Attributes["href"] != null || maximumExclusiveNode.Attributes["id"] != null) 
                {
                    if (maximumExclusiveNode.Attributes["id"] != null) 
                    {
                        maximumExclusiveIDRef_ = maximumExclusiveNode.Attributes["id"].Value;
                        PositiveDecimal ob = new PositiveDecimal(maximumExclusiveNode);
                        IDManager.SetID(maximumExclusiveIDRef_, ob);
                    }
                    else if (maximumExclusiveNode.Attributes["href"] != null)
                    {
                        maximumExclusiveIDRef_ = maximumExclusiveNode.Attributes["href"].Value;
                    }
                    else
                    {
                        maximumExclusive_ = new PositiveDecimal(maximumExclusiveNode);
                    }
                }
                else
                {
                    maximumExclusive_ = new PositiveDecimal(maximumExclusiveNode);
                }
            }
            
        
        }
        
    
        #region MaximumInclusive_
        private PositiveDecimal maximumInclusive_;
        public PositiveDecimal MaximumInclusive_
        {
            get
            {
                if (this.maximumInclusive_ != null)
                {
                    return this.maximumInclusive_; 
                }
                else if (this.maximumInclusiveIDRef_ != null)
                {
                    maximumInclusive_ = IDManager.getID(maximumInclusiveIDRef_) as PositiveDecimal;
                    return this.maximumInclusive_; 
                }
                else
                {
                      return this.maximumInclusive_; 
                }
            }
            set
            {
                if (this.maximumInclusive_ != value)
                {
                    this.maximumInclusive_ = value;
                }
            }
        }
        #endregion
        
        public string maximumInclusiveIDRef_ { get; set; }
        #region MaximumExclusive_
        private PositiveDecimal maximumExclusive_;
        public PositiveDecimal MaximumExclusive_
        {
            get
            {
                if (this.maximumExclusive_ != null)
                {
                    return this.maximumExclusive_; 
                }
                else if (this.maximumExclusiveIDRef_ != null)
                {
                    maximumExclusive_ = IDManager.getID(maximumExclusiveIDRef_) as PositiveDecimal;
                    return this.maximumExclusive_; 
                }
                else
                {
                      return this.maximumExclusive_; 
                }
            }
            set
            {
                if (this.maximumExclusive_ != value)
                {
                    this.maximumExclusive_ = value;
                }
            }
        }
        #endregion
        
        public string maximumExclusiveIDRef_ { get; set; }
        public string choiceStr_0;
        
    
        
    
    }
    
}

File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-fx-5-4/UpperBound.cpp
// UpperBound.cpp 
#include "UpperBound.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

UpperBound::UpperBound(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //maximumInclusiveNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* maximumInclusiveNode = xmlNode->FirstChildElement("maximumInclusive");

   if(maximumInclusiveNode){maximumInclusiveIsNull_ = false;}
   else{maximumInclusiveIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- maximumInclusiveNode , address : " << maximumInclusiveNode << std::endl;
   #endif
   if(maximumInclusiveNode)
   {
      if(maximumInclusiveNode->Attribute("href") || maximumInclusiveNode->Attribute("id"))
      {
          if(maximumInclusiveNode->Attribute("id"))
          {
             maximumInclusiveIDRef_ = maximumInclusiveNode->Attribute("id");
             maximumInclusive_ = boost::shared_ptr<PositiveDecimal>(new PositiveDecimal(maximumInclusiveNode));
             maximumInclusive_->setID(maximumInclusiveIDRef_);
             IDManager::instance().setID(maximumInclusiveIDRef_,maximumInclusive_);
          }
          else if(maximumInclusiveNode->Attribute("href")) { maximumInclusiveIDRef_ = maximumInclusiveNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { maximumInclusive_ = boost::shared_ptr<PositiveDecimal>(new PositiveDecimal(maximumInclusiveNode));}
   }

   //maximumExclusiveNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* maximumExclusiveNode = xmlNode->FirstChildElement("maximumExclusive");

   if(maximumExclusiveNode){maximumExclusiveIsNull_ = false;}
   else{maximumExclusiveIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- maximumExclusiveNode , address : " << maximumExclusiveNode << std::endl;
   #endif
   if(maximumExclusiveNode)
   {
      if(maximumExclusiveNode->Attribute("href") || maximumExclusiveNode->Attribute("id"))
      {
          if(maximumExclusiveNode->Attribute("id"))
          {
             maximumExclusiveIDRef_ = maximumExclusiveNode->Attribute("id");
             maximumExclusive_ = boost::shared_ptr<PositiveDecimal>(new PositiveDecimal(maximumExclusiveNode));
             maximumExclusive_->setID(maximumExclusiveIDRef_);
             IDManager::instance().setID(maximumExclusiveIDRef_,maximumExclusive_);
          }
          else if(maximumExclusiveNode->Attribute("href")) { maximumExclusiveIDRef_ = maximumExclusiveNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { maximumExclusive_ = boost::shared_ptr<PositiveDecimal>(new PositiveDecimal(maximumExclusiveNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<PositiveDecimal> UpperBound::getMaximumInclusive()
{
   if(!this->maximumInclusiveIsNull_){
        if(maximumInclusiveIDRef_ != ""){
             return boost::shared_static_cast<PositiveDecimal>(IDManager::instance().getID(maximumInclusiveIDRef_));
        }else{
             return this->maximumInclusive_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<PositiveDecimal>();
   }
}
boost::shared_ptr<PositiveDecimal> UpperBound::getMaximumExclusive()
{
   if(!this->maximumExclusiveIsNull_){
        if(maximumExclusiveIDRef_ != ""){
             return boost::shared_static_cast<PositiveDecimal>(IDManager::instance().getID(maximumExclusiveIDRef_));
        }else{
             return this->maximumExclusive_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<PositiveDecimal>();
   }
}
}

